diff -u -r rhide-1.5.9.0/idecomp.cc rhide-1.5.9.0-new/idecomp.cc
--- rhide-1.5.9.0/idecomp.cc	2003-03-21 13:53:10.000000000 +0100
+++ rhide-1.5.9.0-new/idecomp.cc	2015-07-01 08:57:41.000000000 +0200
@@ -950,7 +950,7 @@
      */
     for (tok = strtok(dir, " "); tok; tok = strtok(NULL, " "))
     {
-      res = strstr(depfile, tok);
+      res = strstr((char*)depfile, tok);
       if (res == depfile)
         break;
     }
diff -u -r rhide-1.5.9.0/idemak.cc rhide-1.5.9.0-new/idemak.cc
--- rhide-1.5.9.0/idemak.cc	2004-06-18 15:14:23.000000000 +0200
+++ rhide-1.5.9.0-new/idemak.cc	2015-07-01 08:58:04.000000000 +0200
@@ -140,7 +140,7 @@
 
   while (1)
   {
-    lf = strchr(s, '\n');
+    lf = strchr((char*)s, '\n');
     if (lf)
       *lf = 0;
     len = strlen(s);
Only in rhide-1.5.9.0-new/libgdb: libgdbrh.a
diff -u -r rhide-1.5.9.0/libide/ideclass.cc rhide-1.5.9.0-new/libide/ideclass.cc
--- rhide-1.5.9.0/libide/ideclass.cc	2001-12-07 16:41:12.000000000 +0100
+++ rhide-1.5.9.0-new/libide/ideclass.cc	2015-07-01 08:56:17.000000000 +0200
@@ -33,7 +33,7 @@
 {
   if (!_ext)
     return FILE_NOTHING;
-  char *ext = strrchr(_ext, '.');
+  char *ext = strrchr((char*)_ext, '.');
 
   if (!ext)
   {
diff -u -r rhide-1.5.9.0/libide/tfname.cc rhide-1.5.9.0-new/libide/tfname.cc
--- rhide-1.5.9.0/libide/tfname.cc	2001-01-08 22:55:33.000000000 +0100
+++ rhide-1.5.9.0-new/libide/tfname.cc	2015-07-01 08:57:06.000000000 +0200
@@ -79,7 +79,7 @@
   FileEntry fe;
 
   fe.name = (char *) name;
-  fe.slash = strrchr(name, '/');
+  fe.slash = strrchr((char*)name, '/');
   if (!fe.slash)
     fe.slash = fe.name;
   else
diff -u -r rhide-1.5.9.0/rhgdb/gdbsynt.cc rhide-1.5.9.0-new/rhgdb/gdbsynt.cc
--- rhide-1.5.9.0/rhgdb/gdbsynt.cc	2001-12-02 14:35:33.000000000 +0100
+++ rhide-1.5.9.0-new/rhgdb/gdbsynt.cc	2015-07-01 09:00:05.000000000 +0200
@@ -53,7 +53,7 @@
 //                              *TCEditor::SHLSOStack,TCEditor::SHLGenList,
                               TCEditor::SHLGenList, TCEditor::SHLCant) != 0)
   {
-    syntax_file = "__syntax__";
+    syntax_file = (char*)"__syntax__";
     FILE *f = fopen(syntax_file, "w+t");
 
     fprintf(f, "\n\
Only in rhide-1.5.9.0-new/rhgdb: gdbsynt.o
Only in rhide-1.5.9.0-new/rhgdb: main.o
Only in rhide-1.5.9.0-new/rhgdb: options.o
Only in rhide-1.5.9.0-new/rhgdb: rhgdb.exe
diff -u -r rhide-1.5.9.0/rhgdb/rhgdb.mak rhide-1.5.9.0-new/rhgdb/rhgdb.mak
--- rhide-1.5.9.0/rhgdb/rhgdb.mak	2004-06-19 17:49:36.000000000 +0200
+++ rhide-1.5.9.0-new/rhgdb/rhgdb.mak	2015-07-01 09:04:16.000000000 +0200
@@ -47,7 +47,7 @@
 	$(SETOBJ) $(PCRE_OBJ)
 C_DEBUG_FLAGS=-g
 C_OPT_FLAGS=-O2
-C_WARN_FLAGS=-Wall -Werror
+C_WARN_FLAGS=-Wall
 C_C_LANG_FLAGS=
 C_CXX_LANG_FLAGS=
 C_P_LANG_FLAGS=
