diff -r -u rhide-1.5.9.0/idecomp.cc rhide-1.5.9.0-new/idecomp.cc
--- rhide-1.5.9.0/idecomp.cc	2003-03-21 13:53:10.000000000 +0100
+++ rhide-1.5.9.0-new/idecomp.cc	2015-06-30 14:34:03.000000000 +0200
@@ -950,7 +950,7 @@
      */
     for (tok = strtok(dir, " "); tok; tok = strtok(NULL, " "))
     {
-      res = strstr(depfile, tok);
+      res = strstr((char*)depfile, tok);
       if (res == depfile)
         break;
     }
diff -r -u rhide-1.5.9.0/idemak.cc rhide-1.5.9.0-new/idemak.cc
--- rhide-1.5.9.0/idemak.cc	2004-06-18 15:14:23.000000000 +0200
+++ rhide-1.5.9.0-new/idemak.cc	2015-06-30 14:34:28.000000000 +0200
@@ -140,7 +140,7 @@
 
   while (1)
   {
-    lf = strchr(s, '\n');
+    lf = strchr((char*)s, '\n');
     if (lf)
       *lf = 0;
     len = strlen(s);
Only in rhide-1.5.9.0-new/libgdb: echo_ver.exe
Only in rhide-1.5.9.0-new/libgdb: gdb_ver.txt
Only in rhide-1.5.9.0-new/libgdb: libgdbrh.a
Only in rhide-1.5.9.0-new/libgdb: libgdbrh.c
Only in rhide-1.5.9.0-new/libgdb: libgdbrh.h
diff -r -u rhide-1.5.9.0/libide/ideclass.cc rhide-1.5.9.0-new/libide/ideclass.cc
--- rhide-1.5.9.0/libide/ideclass.cc	2001-12-07 16:41:12.000000000 +0100
+++ rhide-1.5.9.0-new/libide/ideclass.cc	2015-06-30 14:30:21.000000000 +0200
@@ -33,7 +33,7 @@
 {
   if (!_ext)
     return FILE_NOTHING;
-  char *ext = strrchr(_ext, '.');
+  char *ext = strrchr((char*)_ext, '.');
 
   if (!ext)
   {
diff -r -u rhide-1.5.9.0/libide/tflagent.cc rhide-1.5.9.0-new/libide/tflagent.cc
--- rhide-1.5.9.0/libide/tflagent.cc	2001-01-08 22:55:33.000000000 +0100
+++ rhide-1.5.9.0-new/libide/tflagent.cc	2015-06-30 14:31:17.000000000 +0200
@@ -3,6 +3,7 @@
 #define Uses_TDialog
 #define Uses_TRect
 #define Uses_TValidator
+#define Uses_TRangeValidator
 #define Uses_TInputLine
 #define Uses_TProgram
 #define Uses_TApplication
diff -r -u rhide-1.5.9.0/libide/tfname.cc rhide-1.5.9.0-new/libide/tfname.cc
--- rhide-1.5.9.0/libide/tfname.cc	2001-01-08 22:55:33.000000000 +0100
+++ rhide-1.5.9.0-new/libide/tfname.cc	2015-06-30 14:31:40.000000000 +0200
@@ -79,7 +79,7 @@
   FileEntry fe;
 
   fe.name = (char *) name;
-  fe.slash = strrchr(name, '/');
+  fe.slash = strrchr((char*)name, '/');
   if (!fe.slash)
     fe.slash = fe.name;
   else
diff -r -u rhide-1.5.9.0/librhuti/io.cc rhide-1.5.9.0-new/librhuti/io.cc
--- rhide-1.5.9.0/librhuti/io.cc	2001-09-01 23:56:27.000000000 +0200
+++ rhide-1.5.9.0-new/librhuti/io.cc	2015-06-30 14:43:11.000000000 +0200
@@ -16,6 +16,11 @@
 static int h_out, h_outbak;
 static int h_err, h_errbak;
 
+/* hack for setedit */
+char *unique_name(const char *before, char *retval){
+     return unique_name( (char*) before,retval);
+}
+
 /* returns a malloced unique tempname in $TMPDIR */
 char *
 unique_name(char *before, char *retval)
diff -r -u rhide-1.5.9.0/libtvuti/bigmessa.cc rhide-1.5.9.0-new/libtvuti/bigmessa.cc
--- rhide-1.5.9.0/libtvuti/bigmessa.cc	2001-10-17 21:11:58.000000000 +0200
+++ rhide-1.5.9.0-new/libtvuti/bigmessa.cc	2015-06-30 14:32:47.000000000 +0200
@@ -136,7 +136,7 @@
   r = TRect(4, 3, dialog->size.x - 3, 4);
 //    r = TRect( 4 + strlen(aLabel), 2, dialog->size.x - 3, 3 );
   control = new TInputLine(r, limit);
-  ((TInputLine *) control)->SetValidator(validator);
+  ((TInputLine *) control)->setValidator(validator);
   dialog->insert(control);
 
   r = TRect(2, 2, 3 + strlen(aLabel), 3);
diff -r -u rhide-1.5.9.0/libtvuti/tintinpu.cc rhide-1.5.9.0-new/libtvuti/tintinpu.cc
--- rhide-1.5.9.0/libtvuti/tintinpu.cc	2001-01-08 22:55:35.000000000 +0100
+++ rhide-1.5.9.0-new/libtvuti/tintinpu.cc	2015-06-30 14:33:16.000000000 +0200
@@ -14,9 +14,9 @@
 TEnterInputLine(bounds, aMaxlen)
 {
   if (_unsigned)
-    SetValidator(new TRangeValidator(0, UINT_MAX));
+    setValidator(new TRangeValidator(0, UINT_MAX));
   else
-    SetValidator(new TRangeValidator(INT_MIN, INT_MAX));
+    setValidator(new TRangeValidator(INT_MIN, INT_MAX));
 }
 
 void

--- rhide-1.5.9.0/idemain.cc.orig	2015-06-30 15:08:31.000000000 +0200
+++ rhide-1.5.9.0/idemain.cc	2015-06-30 15:09:11.000000000 +0200
@@ -37,6 +37,7 @@
 #define Uses_TFileDialog
 #define Uses_TFileInputLine
 #define Uses_TValidator
+#define Uses_TRangeValidator
 #define Uses_TDirList
 #define Uses_TButton
 #define Uses_TEventQueue
--- rhide-1.5.9.0/idesynta.cc.orig	2015-06-30 15:09:51.000000000 +0200
+++ rhide-1.5.9.0/idesynta.cc	2015-06-30 15:10:17.000000000 +0200
@@ -668,7 +668,7 @@
   r1.b.y = r1.a.y + 1;
   r1.b.x = r1.a.x + 7;
   inp = new TIntInputLine(r1, 4);
-  inp->SetValidator(new TRangeValidator(1, 1000));
+  inp->setValidator(new TRangeValidator(1, 1000));
   r = r1;
   r1.move(0, -1);
   tmp = _("Fu~z~zy value");
